luda-adventure/src/app/game/render/render_interaction_guide.rs

```diff
@@ -0,0 +1,80 @@
+use crate::app::game::{
+    nearest_entity_id, Game, GameState, RenderingContext, MAX_INTERACTION_DISTANCE,
+};
+use namui::prelude::*;
+
+const ICON_SIZE: Px = px(36.0);
+const OFFSET_Y: Px = px(-4.0);
+
+impl Game {
+    pub fn render_interaction_guide(
+        &self,
+        game_state: &GameState,
+        rendering_context: &RenderingContext,
+    ) -> RenderingTree {
+        let interactive_object_list = self.get_interactive_object_with_distance(game_state);
+        let Some(nearest_entity_id) = nearest_entity_id(&interactive_object_list) else {
+            return RenderingTree::Empty;
+        };
+        render(interactive_object_list.into_iter().filter_map(
+            |((entity, (positioner, renderer)), distance)| {
+                let xy = positioner.xy_with_interpolation(rendering_context.interpolation_progress);
+                let visual_rect = renderer.visual_rect() + xy;
+                let icon_xy = Xy {
+                    x: rendering_context.px_per_tile
+                        * (visual_rect.left() + visual_rect.right())
+                        * 0.5,
+                    y: rendering_context.px_per_tile * visual_rect.y() + OFFSET_Y
+                        - (ICON_SIZE * 0.5),
+                };
+                let scale = 1. - (distance / MAX_INTERACTION_DISTANCE).max(0.).min(1.);
+                if scale < 0.01 {
+                    return None;
+                }
+                Some(interaction_icon(
+                    icon_xy,
+                    scale,
+                    entity.id() == nearest_entity_id,
+                ))
+            },
+        ))
+    }
+}
+
+fn interaction_icon(xy: Xy<Px>, scale: f32, nearest: bool) -> RenderingTree {
+    let scale = match nearest {
+        true => scale * 1.2,
+        false => scale,
+    };
+    text(TextParam {
+        text: "press Z".to_string(),
+        x: xy.x,
+        y: xy.y,
+        align: TextAlign::Center,
+        baseline: TextBaseline::Middle,
+        font_type: FontType {
+            serif: false,
+            size: (ICON_SIZE * scale).into(),
```

Comment:
This will make a problem.
Font is (Typeface + FontSize). So, when you make a variable font size with scale, it will make many fonts. It wouldn't that much cheap.

I think you should namui::scale for lag.